CMakeLists.txt

```diff
@@ -930,16 +930,32 @@ add_custom_target(lrelease
 
 add_dependencies(mscore pluginDocumentation)
 
-
 ##
-##  For MSVC: set the startup project to be mscore
-##     (requires CMake 3.6.3+ to work, but should be benign on other versions
-##      as it is just setting the value for a property).
+## Miscellaneous Microsoft Visual Studio settings
 ##
 if (MSVC)
+
+   # Force the "install" and "package" targets not to depend on the "all" target.
    set(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY true)
    set(CMAKE_SKIP_PACKAGE_ALL_DEPENDENCY true)
-   set(VS_STARTUP_PROJECT mscore)
+
+   # Set the startup project to "mscore".
+   if (NOT ${CMAKE_VERSION} VERSION_LESS "3.6.0")
+      set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT mscore)
+   endif ()
+
+   # Set the debugging properties for the "mscore" project.
+   file(TO_NATIVE_PATH "${CMAKE_INSTALL_PREFIX}/bin" VS_DEBUGGER_WORKING_DIRECTORY)
+   if (NOT ${CMAKE_VERSION} VERSION_LESS "3.12.0")
```

Comment:
3.12, then 3.8, then 3.13?
I there a logic I don't get here?

Reply:
Support for each of those individual settings was added in different versions of CMake.

Reply:
Why not go for chronological order then?

Reply:
They're arranged in logical order, which is the same order in which they appear in the VS UI.

The order in which they started being supported in CMake is arbitrary (and unimportant), but a version check does still need to be done for each one.

Reply:
Ok thanks. I get it.